tests/fixtures/logging.py

```diff
@@ -5,14 +5,14 @@
 
 
 @pytest.fixture(autouse=True)
-def reset_api_log_handler():
+async def reset_api_log_handler():
     """
     Since we have a singleton worker for the runtime of the process, we must reset
     it to `None` between tests and we should flush logs before each test exits to
     stop the logging thread.
     """
     yield
-    APILogHandler.flush()
+    await APILogHandler.aflush()
```

Comment:
just curious, why the switch to async here?

Reply:
I changed `flush` to not be a classmethod to override the base class in an incompatible way. `aflush` is still a classmethod, so I'm using it here to keep the fixture working the same way.

Reply:
aha i see i see, thanks!